src/figuresClasses.ts

```diff
@@ -1,19 +1,84 @@
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' |'blue';
+
 export interface Figure {
+  shape: Shape,
+  color: Color,
 
+  getArea: () => number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape: Shape = 'triangle';
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      // eslint-disable-next-line
+      throw new Error(`Every side (a: ${a}, b: ${b}, c: ${c})of triangle must be greater than 0`);
+    }
+
+    if (a >= b + c || b >= a + c || c >= a + b) {
+    // eslint-disable-next-line
+      throw new Error('The longest side of a triangle should be greater or equal to a sum of two others');
+    }
+  }
 
+  getArea(): number {
+    const halfOfAPerimeter = (this.a + this.b + this.c) / 2;
+    const square = Math.floor(
+      Math.sqrt(halfOfAPerimeter
+        * (halfOfAPerimeter - this.a)
+        * (halfOfAPerimeter - this.b)
+        * (halfOfAPerimeter - this.c)) * 100,
+    ) / 100;
+
+    return square;
```

Comment:
DRY - don't repeat yourself
move this rounding to the separate function
```suggestion
    const square = Math.sqrt(halfOfAPerimeter
        * (halfOfAPerimeter - this.a)
        * (halfOfAPerimeter - this.b)
        * (halfOfAPerimeter - this.c));

    return getRounding(square);
```